model/infinispan/src/main/java/org/keycloak/models/sessions/infinispan/remotestore/RemoteCacheInvoker.java

```diff
@@ -108,7 +114,24 @@ private <K, V extends SessionEntity> void runOnRemoteCache(TopologyInfo topology
 
         switch (operation) {
             case REMOVE:
-                remoteCache.remove(key);
+                if (Profile.isFeatureEnabled(Profile.Feature.PERSISTENT_USER_SESSIONS) &&
```

Comment:
It is not required since removal is always replicated to the remote site.

Reply:
Resolved in next commit